R/mstate.R

```diff
@@ -827,78 +827,87 @@ form.basepars.tcovs <- function(x, transi, # index of allowed transition
 ##' @export
 sim.fmsm <- function(x, trans=NULL, t, newdata=NULL, start=1, M=10, tvar="trans", tcovs=NULL, tidy=FALSE, debug=FALSE){
     if (is.null(trans)) {
-        if (!is.null(attr(x, "trans"))) trans <- attr(x, "trans")
+        if (!is.null(attr(x, "trans"))) 
+            trans <- attr(x, "trans")
         else stop("`trans` not supplied and not found in `x`")
     }
-    if (length(t)==1) t <- rep(t, M)
-    else if (length(t)!=M) stop("length of t should be 1 or M=",M)
-    if (length(start)==1) start <- rep(start, M)
-    else if (length(start)!=M) stop("length of start should be 1 or M=",M)
-
-    basepars.all <- pars.fmsm(x=x, trans=trans, newdata=newdata, tvar=tvar)
-
-    nst <- nrow(trans)
-    ## TODO only need a max time if model is transient, else if absorbing, can allocate these up front
-    res.st <- cur.st <- start
-    res.t <- cur.t <- rep(0, M)
-    todo <- seq_len(M)
-    while (any(todo)){
-        if (debug) { cat("cur.t\n"); cat(cur.t); cat("\n") }
-        if (debug) { cat("cur.st\n"); cat(cur.st); cat("\n") }
-        if (debug) { cat("TODO\n"); cat(todo); cat("\n") }
+    if (length(t) == 1) 
+        t <- rep(t, M)
+    else if (length(t) != M) 
+        stop("length of t should be 1 or M=", M)
+    if (length(start) == 1) 
+        start <- rep(start, M)
+    else if (length(start) != M) 
+        stop("length of start should be 1 or M=", M)
+    basepars.all <- flexsurv:::pars.fmsm(x = x, trans = trans, newdata = newdata, 
```

Comment:
we could optionally presample a bootstrap sample of size M here to allow simulation under the asymptotic normal distribution of the parameters. This would be very useful to compute all quantities of interest form the resulting simulated results.